### PR TITLE
Module4

### PR DESCRIPTION
**Pull request by ArturEpam**
Implemented module 4 and added environment variables for configuration.
I didn't add that 'deletedAt' because I run out of time for more refactoring :/.
Split schema and data initialization in DbInitializer.

Thanks :)

### CODE REVIEW

File: src/data-access/UserGroupRepository.ts [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+@Service()
+export class UserGroupRepository {
+    private sequelize: Sequelize;
+
+    constructor(@Inject('database') sequelize: Sequelize) {
+        this.sequelize = sequelize;
+    }
+
+    public async createAll(userGroups: UserGoupAttributes[]): Promise<UserGroup[]> {
+        return await this.sequelize.transaction(async (t) => {
```
Comment (thebergamo) [inline src/data-access/UserGroupRepository.ts]: You don't need to use `await` when you're returning the function directly.
So you can remove it in all the cases similar.

File: src/data-access/UserRepository.ts [hunk 2]
```diff
@@ -9,12 +9,12 @@ export class UserRepository {
     private sequelize: Sequelize;
 
     constructor(@Inject('database') sequelize: Sequelize) {
-        User.initialize(sequelize);
         this.sequelize = sequelize;
     }
 
     public async findById(id: string): Promise<User> {
-        return await User.findOne({ where: { ...this.notDeleted, id } });
+        const user = await User.findOne({ where: { ...this.notDeleted, id }, include: 'groups' });
```
Comment (thebergamo) [inline src/data-access/UserRepository.ts]: in this case you can return directly unless you want to use it

**Review by thebergamo — COMMENTED** [review]
Besides my minor comments. Nothing else I can comment

File: src/data-access/UserGroupRepository.ts (hunk 1 above)
Comment (ArturEpam) [reply src/data-access/UserGroupRepository.ts]: right, I'll fix it.

**Pull request merged**